### ISSUE TITLE
`inputText`: add a way to obscure sensitive text

### ISSUE THREAD

**Issue by HenSquared** [mobile-dev-inc/maestro#1226]
**Is your feature request related to a problem? Please describe.**

Log outputs include all inputText. If a flow includes a login, the resulting maestro.log will expose that password. 

**Describe the solution you'd like**

Perhaps there should be an API: `inputSensitiveText: password`

If this is the case, the value of the field should be omitted or censored (XXXXXXXXX) in logging. 

For example if I have a sensitive text of value: Password1, I currently input it using this:

```console
maestro test login.yml --env=PASSWORD=Password1
```

**test.yaml**

```yaml
...
inputText: ${PASSWORD}
...

```
and logging outputs this as:

```
[INFO ] m.cli.runner.MaestroCommandRunner - Input text ${PASSWORD} metadata CommandMetadata(numberOfRuns=null, evaluatedCommand=MaestroCommand(tapOnElement=null, tapOnPoint=null, tapOnPointV2Command=null, scrollCommand=null, swipeCommand=null, backPressCommand=null, assertCommand=null, assertConditionCommand=null, inputTextCommand=InputTextCommand(text=Password1), inputRandomTextCommand=null, launchAppCommand=null, applyConfigurationCommand=null, openLinkCommand=null, pressKeyCommand=null, eraseTextCommand=null, hideKeyboardCommand=null, takeScreenshotCommand=null, stopAppCommand=null, clearStateCommand=null, clearKeychainCommand=null, runFlowCommand=null, setLocationCommand=null, repeatCommand=null, copyTextCommand=null, pasteTextCommand=null, defineVariablesCommand=null, runScriptCommand=null, waitForAnimationToEndCommand=null, evalScriptCommand=null, mockNetworkCommand=null, scrollUntilVisible=null, travelCommand=null, assertOutgoingRequestsCommand=null, startRecordingCommand=null, stopRecordingCommand=null), logMessages=[])
[INFO ] maestro.Maestro - Inputting text: Password1
```

Perhaps we can update API and logging to look more like this:

```
maestro test login.yml --env=PASSWORD=Password1
```

**test.yaml**

```yaml
...
inputSensitiveText: ${PASSWORD}
...

```
and logging outputs this as:

```
[INFO ] m.cli.runner.MaestroCommandRunner - Input text ${PASSWORD} metadata CommandMetadata(numberOfRuns=null, evaluatedCommand=MaestroCommand(tapOnElement=null, tapOnPoint=null, tapOnPointV2Command=null, scrollCommand=null, swipeCommand=null, backPressCommand=null, assertCommand=null, assertConditionCommand=null, inputTextCommand=InputTextCommand(text=XXXX), inputRandomTextCommand=null, launchAppCommand=null, applyConfigurationCommand=null, openLinkCommand=null, pressKeyCommand=null, eraseTextCommand=null, hideKeyboardCommand=null, takeScreenshotCommand=null, stopAppCommand=null, clearStateCommand=null, clearKeychainCommand=null, runFlowCommand=null, setLocationCommand=null, repeatCommand=null, copyTextCommand=null, pasteTextCommand=null, defineVariablesCommand=null, runScriptCommand=null, waitForAnimationToEndCommand=null, evalScriptCommand=null, mockNetworkCommand=null, scrollUntilVisible=null, travelCommand=null, assertOutgoingRequestsCommand=null, startRecordingCommand=null, stopRecordingCommand=null), logMessages=[])
[INFO ] maestro.Maestro - Inputting text: XXXX
```

**Describe alternatives you've considered**

For my usecase, I am using GitHub actions and archiving the resulting failure logs from maestro. As a workaround, I can run some find and replace for any sensitive text before outputting the archive file.

**Comment by Fishbowler** [comment]
Labelling implemented in #1292 will offer (when released) a workaround that allows you to specify a label in your yaml that should replace the text at the beginning of the first of these log lines (but it won't affect the second 😞)

**Comment by Fishbowler** [comment]
I think this can be closed as solved by #1481?

**Comment by TheAlmightyBob** [comment]
@Fishbowler Can you give an example of how that PR would address this? Your own previous comment said
> should replace the text at the beginning of the first of these log lines (but it won't affect the second 😞)

implying that you didn't think the PR would fully resolve this.

**Comment by Fishbowler** [comment]
Ah, yep, you're 100% correct. This is only half fixed.

**Comment by bartekpacia** [comment]
Perhaps instead of adding a new `inputSensitiveText` command, we could add a new boolean argument to `inputText`?

```yaml
- inputText:
    text: Hello World
    mask: true
```

**Comment by Fishbowler** [comment]
How would we expect this to interact with the debug JSON outputter?
Preserve the string in this instance?

**Comment by StevenMartinMedgate** [comment]
Is this still something that is tracked and planned as feature? Would be nice since this is the only thing preventing us from using maestro with browserstack's integration as it currently seems that browserstack itself does have redact feature for other frameworks (appium/wdio) which we are currently using, but not for maestro.

**Comment by Fishbowler** [comment]
@StevenMartinMedgate 

Tracked: clearly, it's right here :)
Planned: The beauty of open source isn't that it's zero cost, it's that anyone can see the code, and anyone can contribute changes they want. 

On the Browserstack thing, they're running their own fork I've heard, so I wouldn't be confident in a fix here necessarily making your experience there any different. Their fork has other changes though - forks are another great thing about open source! But it'd help you if they contributed their changes upstream, I guess.

TL;DR This isn't something staff are active on right now.

**Comment by Mohanad49** [comment]
@Fishbowler picking up your offer from above — I've resubmitted this as #3496.

It's @tokou's #1944 rebased onto current `main` (which had moved 559 commits), with @bartekpacia's design from this thread. Credit for the approach and the original implementation is theirs.

Rebasing surfaced three things that were in the original branch rather than caused by the drift, and one is worth flagging here rather than only in the PR:

**`redact: true` silently made the command optional.** `YamlInputText.parse` passed its values positionally into a constructor whose third parameter is `optional`:

```kotlin
data class YamlInputText(text, label, optional, redact)
return YamlInputText(input, label, redact)   // redact -> optional
```

Both are `Boolean`, so it compiled and the tests that existed didn't cover it. The effect would have been that a failing `inputText` step got skipped instead of failing the flow — which seems worse than the logging problem this issue is about. Fixed by passing named arguments.

The other two: `getOrDefault("redact", "false") as Boolean` threw `ClassCastException` whenever the key was absent, and the custom serializer omitted `optional`, which broke the existing `serialize InputTextCommand` test.

The branch also passed `redact` into `InputRandomCommand`, which has no such field, so it didn't compile. I scoped the PR to `inputText` only, matching this issue — happy to extend it to `inputRandom*` separately if you'd like that too.

**One open question:** this thread suggests `mask: true` and #1944 implemented `redact: true`. I kept `redact` to stay close to the branch that was already reviewed, but I have no attachment to it — say the word and I'll rename it.

I've added an e2e flow using the existing Input/Keyboard screen, so no demo-app change was needed. Its assertion checks that the *real* text still reaches the app, since redaction should change what Maestro reports and never what it types. I couldn't run the e2e suite locally (no device), so I flagged that flow as unverified when I first posted this. CI has since run it on **both Android and iOS** and it passes, so that caveat no longer applies.

All 14 checks on #3496 are green: unit tests on Java 17 (449 across `maestro-orchestra` and `maestro-orchestra-models`), builds on Linux/macOS/Windows, and the Android, iOS and Web e2e suites.
